File: Programmers/숫자 문자열과 영단어.py
# 2021 카카오 채용연계형 인턴십

def solution(s):
    answer = ""
    temp = ""
    convert = {"zero": "0", "one": "1", "two": "2", "three": "3", "four": "4",
              "five": "5", "six": "6", "seven": "7", "eight": "8", "nine": "9"}
    
    for i in s:
        if temp in convert.keys():
            answer += convert[temp]
            temp = ""
        if i.isalpha():
            temp += i
        else:
            answer += i
    
    if temp:
        answer += convert[temp]
    
    answer = int(answer)
    
    return answer

File: Programmers/숫자 문자열과 영단어.c
#include <stdio.h>
#include <string.h>
#include <stdlib.h>

int convertWordToDigit(char *word) {
    if (strcmp(word, "zero") == 0) return 0;
    if (strcmp(word, "one") == 0) return 1;
    if (strcmp(word, "two") == 0) return 2;
    if (strcmp(word, "three") == 0) return 3;
    if (strcmp(word, "four") == 0) return 4;
    if (strcmp(word, "five") == 0) return 5;
    if (strcmp(word, "six") == 0) return 6;
    if (strcmp(word, "seven") == 0) return 7;
    if (strcmp(word, "eight") == 0) return 8;
    if (strcmp(word, "nine") == 0) return 9;
    return -1;
}

int solution(const char *s) {
    char answer[50] = "";
    char temp[10] = "";
    int answerIndex = 0;

    for (int i = 0; i < strlen(s); i++) {
        if (isalpha(s[i])) {
            strncat(temp, &s[i], 1);
            int digit = convertWordToDigit(temp);
            if (digit != -1) {
                answer[answerIndex++] = '0' + digit;
                temp[0] = '\0';
            }
        } else {
            answer[answerIndex++] = s[i];
        }
    }

    if (strlen(temp) > 0) {
        int digit = convertWordToDigit(temp);
        if (digit != -1) {
            answer[answerIndex++] = '0' + digit;
        }
    }

    answer[answerIndex] = '\0';

    return atoi(answer);
}

int main() {
    const char *input = "one4seveneight";
    int result = solution(input);
    printf("%d\n", result);
    return 0;
}
